src/server/components/lists/searches/translators-interpreters.ts

```diff
@@ -68,116 +38,35 @@ function hasSworn(results: TranslatorInterpreterListItemGetObject[]): SwornOutpu
   };
 }
 
-export async function searchTranslatorsInterpreters(req: Request, res: Response): Promise<void> {
-  const params = getAllRequestParams(req);
-  const { serviceType, country, region, print = "no" } = params;
-  let { page = "1" } = params;
-  page = page !== "" ? page : "1";
-
-  const pageNum = parseInt(page);
-  params.page = pageNum.toString();
+export async function searchTranslatorsInterpreters(req: Request) {
+  const { answers = {} } = req.session;
+  const { country, serviceType } = answers;
+  const { print = "no", page = 1 } = req.query;
+  const pageNum = parseInt(page as string);
 
-  if (!country) {
-    const query = new URLSearchParams(req.query as Record<string, string>);
-    res.redirect(`${listsRoutes.finder}?${query.toString()}`);
-    return;
-  }
-
-  let languageNamesProvided;
-  let serviceNamesProvided;
-  let servicesProvided;
   let allRows: TranslatorInterpreterListItemGetObject[] = [];
   let searchResults: TranslatorInterpreterListItemGetObject[] = [];
-  let filterProps: {
-    countryName: typeof countriesList[number]["value"] | undefined;
-    region?: string;
-    servicesProvided: string[] | undefined;
-    languagesProvided: string[] | undefined;
-    interpreterServices: string[] | undefined;
-    translationSpecialties: string[] | undefined;
-    offset: number;
-  } = {
-    countryName: undefined,
-    region: "",
-    servicesProvided: [],
-    languagesProvided: [],
-    interpreterServices: [],
-    translationSpecialties: [],
+  const filterProps = {
+    countryName: validateCountryLower(country!),
+    region: decodeURIComponent(answers.region ?? ""),
+    servicesProvided: sanitiseServices(answers.services),
+    languagesProvided: sanitiseLanguages(answers.languages),
+    interpreterServices: sanitiseInterpretationTypes(answers.interpretationTypes ?? []),
+    translationSpecialties: sanitiseTranslationTypes(answers.translationTypes ?? []),
     offset: -1,
   };
 
   try {
-    const countryName = validateCountry(country);
```

Comment:
this code was very verbose..

fwiw - `!= null` doesn't behave how one might think. 

- none of these functions actually returned `null`, so it doesn't make sense to check for null
- this code only worked because the return values are actually `undefined` AND != is a non-strict inequality. null and undefined are treated as the same